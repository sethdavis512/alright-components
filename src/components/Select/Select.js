import React from 'react';
import PropTypes from 'prop-types';

function Select({ label, onChange, options }) {
    const mappedOptions = options.map(option => (
        <option key={option.value} value={option.value}>
            {option.label}
        </option>
    ));
    return (
        <label>
            {label}
            <select onChange={onChange}>{mappedOptions}</select>
        </label>
    );
}

Select.propTypes = {
    onChange: PropTypes.func.isRequired,
    options: PropTypes.array.isRequired
};

Select.defaultProps = {
    options: [{ label: 'Option 1', value: 1 }]
};

export default Select;
