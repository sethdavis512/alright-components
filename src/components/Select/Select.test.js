import React from 'react';
import { shallow } from 'enzyme';
import Select from './Select';

describe('Select', () => {
    const defaultProps = {
        onChange: () => {},
        options: [
            { label: 'Red', value: 'red' },
            { label: 'Green', value: 'green' },
            { label: 'Blue', value: 'blue' }
        ],
        value: 'Green'
    };

    function renderComponent(props) {
        return shallow(<Select {...defaultProps} {...props} />);
    }

    it('should match snapshot', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });
});
