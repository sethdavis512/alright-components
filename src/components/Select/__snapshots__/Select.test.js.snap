// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
<label>
  <select
    onChange={[Function]}
  >
    <option
      key="red"
      value="red"
    >
      Red
    </option>
    <option
      key="green"
      value="green"
    >
      Green
    </option>
    <option
      key="blue"
      value="blue"
    >
      Blue
    </option>
  </select>
</label>
`;
